src/SFA.DAS.ApplyService.Web/Authorisation/AuthorizationHandler.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.ApplyService.Domain.Entities;
+using SFA.DAS.ApplyService.Web.Infrastructure;
+
+namespace SFA.DAS.ApplyService.Web.Authorization
+{
+    public class AuthorizationHandler : IAuthorizationHandler
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IApplicationApiClient _apiClient;
+        private readonly IUserService _userService;
+        private readonly ILogger<AuthorizationHandler> _logger;
+
+        public AuthorizationHandler(IHttpContextAccessor httpContextAccessor,
+            IApplicationApiClient apiClient,
+            IUserService userService,
+            ILogger<AuthorizationHandler> logger)
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _apiClient = apiClient;
+            _userService = userService;
+            _logger = logger;
+        }
+
+        public async Task HandleAsync(AuthorizationHandlerContext context)
+        {
+            _logger.LogInformation("AccessApplicationRequirementHandler invoked");
+
+            var requestedApplicationId = _httpContextAccessor.HttpContext.Request.Query["ApplicationId"].ToString();
```

Comment:
I can't remember but does this return an empty string or null if the QueryString isn't there?